### ISSUE TITLE
VS Code extension implemented by `createAsyncLanguageServicePlugin` is broken when a non-ts file is first opened in VS Code

### ISSUE THREAD

**Issue by mizdra** [volarjs/volar.js#258]
## Summary

I use `createAsyncLanguageServicePlugin` to develop the VS Code extension (`vscode-css-modules`) and TypeScript Language Service Plugin (`typescript-plugin-css-modules`) for CSS Modules.

`vscode-css-modules` makes tsserver load `typescript-plugin-css-modules` by setting `contributes.typescriptServerPlugins`. It also sets `activationEvents` so that tsserver starts when the CSS file is opened. In addition, disable VS Code's built-in CSS extension (`vscode.css-language-features`) so that it do not conflict with `vscode-css-modules`.

`packages/vscode/package.json`:
```json
{
  // ...
  "activationEvents": [
    "onLanguage:css"
  ],
  "contributes": {
    "typescriptServerPlugins": [
      {
        "name": "typescript-plugin-css-modules",
        "configNamespace": "typescript",
        "languages": [
          "css"
        ]
      }
    ],
  // ...
}
```

`packages/vscode/src/index.cjs`:
```js
exports.activate = function activate() {
  console.log('[vscode-css-modules] Activated');

  // By default, `vscode.typescript-language-features` is not activated when a user opens *.css in VS Code.
  // So, activate it manually.
  const tsExtension = vscode.extensions.getExtension('vscode.typescript-language-features');
  if (tsExtension) {
    console.log('[vscode-css-modules] Activating `vscode.typescript-language-features`');
    tsExtension.activate();
  }
  // ...
```

`packages/ts-plugin/src/index.cjs`:
```js
const { createAsyncLanguageServicePlugin } = require('@volar/typescript/lib/quickstart/createAsyncLanguageServicePlugin.js');
const { createCssModuleLanguagePlugin } = require('./language-plugin.cjs');
const ts = require('typescript');

module.exports = createAsyncLanguageServicePlugin(['.css'], ts.ScriptKind.TS, async (ts, info) => {
  return {
    languagePlugins: [createCssModuleLanguagePlugin()],
  };
});
```

`.vscode/launch.json`
```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Start extension in VS Code",
      "type": "extensionHost",
      "request": "launch",
      "args": [
        // ...
        "--disable-extension=vscode.css-language-features",
        // ...
```

This usually works: tsserver has control over the CSS.

https://github.com/user-attachments/assets/a7dfa6bf-76ff-4e7c-8937-4b9e3572ed83

However, when VS Code is launched and the CSS file is first opened, the language feature does not work.

https://github.com/user-attachments/assets/67609435-29bc-42ed-bfe9-683d2f4eb497

## Reproduction

repository: https://github.com/mizdra/repro-volar-open-non-ts-file-first

1. Execute `npm i` in the terminal.
1. Open this workspace in VS Code.
1. Run "Start extension in VS Code" debug task (F5).
1. Hover `a_2` and do `Find all references`

## Expected behavior

`a_2` in `index.ts` appears in the list.

## Actual behavior

`a_2` in `index.ts` does not appear in the list.

## Additional context

This seems to be caused by `createAsyncLanguageServicePlugin`. If you use `createLanguageServicePlugin` instead, the problem does not reproduce.

You can try it by reverting https://github.com/mizdra/repro-volar-open-non-ts-file-first/commit/028198a16190d6b8a8d11ab62216caf4ead09573.

## Environment
- VSCode: 1.96.2
- OS: Darwin arm64 24.1.0
- @volar/typescript: 2.4.11
- @volar/language-core: 2.4.11

**Comment by johnsoncodehk** [comment]
Can you use `jiti` instead of `await import` to allow switch to `createLanguageServicePlugin` API?

See: https://github.com/typed-ember/glint/pull/803

**Comment by mizdra** [comment]
Yes, I think I can use that. However, I have chosen another way around this problem. So I am not currently interested in solving this problem.

I have used `await import('...')` to import ESM from CJS. But then I converted all the code to CJS. The whole code is now synchronous and I can use `createLanguageServicePlugin`.

**Comment by mizdra** [comment]
`createAsyncLanguageServicePlugin` was dropped (https://github.com/volarjs/volar.js/pull/272). So I close this issue :)
